src/main/java/org/jabref/gui/util/FilteredListProxy.java

```diff
@@ -16,15 +16,52 @@
 
 public class FilteredListProxy {
     private static final Logger LOGGER = LoggerFactory.getLogger(FilteredListProxy.class);
-
-    private FilteredListProxy() {
+    private static final Method BEGIN_CHANGE_METHOD;
+    private static final Method END_CHANGE_METHOD;
+    private static final Method NEXT_ADD_METHOD;
+    private static final Method NEXT_UPDATE_METHOD;
+    private static final Method NEXT_REMOVE_METHOD;
+    private static final Method REFILTER_METHOD;
+    private static final Method ENSURE_SIZE_METHOD;
+    private static final Method GET_PREDICATE_IMPL_METHOD;
+    private static final Field FILTERED_FIELD;
+    private static final Field SIZE_FIELD;
+
+    static {
+        try {
+            BEGIN_CHANGE_METHOD = ObservableListBase.class.getDeclaredMethod("beginChange");
+            END_CHANGE_METHOD = ObservableListBase.class.getDeclaredMethod("endChange");
+            NEXT_ADD_METHOD = ObservableListBase.class.getDeclaredMethod("nextAdd", int.class, int.class);
+            NEXT_UPDATE_METHOD = ObservableListBase.class.getDeclaredMethod("nextUpdate", int.class);
+            NEXT_REMOVE_METHOD = ObservableListBase.class.getDeclaredMethod("nextRemove", int.class, Object.class);
+
+            REFILTER_METHOD = FilteredList.class.getDeclaredMethod("refilter");
+            ENSURE_SIZE_METHOD = FilteredList.class.getDeclaredMethod("ensureSize", int.class);
+            GET_PREDICATE_IMPL_METHOD = FilteredList.class.getDeclaredMethod("getPredicateImpl");
+
+            FILTERED_FIELD = FilteredList.class.getDeclaredField("filtered");
+            SIZE_FIELD = FilteredList.class.getDeclaredField("size");
+
+            BEGIN_CHANGE_METHOD.setAccessible(true);
+            END_CHANGE_METHOD.setAccessible(true);
+            NEXT_ADD_METHOD.setAccessible(true);
+            NEXT_UPDATE_METHOD.setAccessible(true);
+            NEXT_REMOVE_METHOD.setAccessible(true);
+
+            REFILTER_METHOD.setAccessible(true);
+            ENSURE_SIZE_METHOD.setAccessible(true);
+            GET_PREDICATE_IMPL_METHOD.setAccessible(true);
+
+            FILTERED_FIELD.setAccessible(true);
+            SIZE_FIELD.setAccessible(true);
+        } catch (NoSuchMethodException | NoSuchFieldException e) {
+            throw new RuntimeException("Failed to initialize reflective methods or fields", e);
+        }
     }
```

Comment:
Im not happy initializing this in a static block. You have no control, when this is done.
Also it extends the loading time of jabref, since it will be sometime in the initialization process.
Maybe the methods can be cached anyway the first time called with some sort of strategy method?

Reply:
Done in: [`3a298ef` (#11624)](https://github.com/JabRef/jabref/pull/11624/commits/3a298efc9c4d792894882e6b2bd1240d8e6e774e).